### ISSUE TITLE
`torch.backends.cudnn.flags(enable=True, benchmark=True)` ctx manager does not work after `torch.backends.disable_global_flags()` is executed

### ISSUE THREAD

**Issue by thiagocrepaldi** [pytorch/pytorch#81908]
### 🐛 Describe the bug

TL;DR; `with torch.backends.cudnn.flags(enable=True, benchmark=True):` ctx manager does not enable CUDNN after `torch.backends.disable_global_flags()` is executed before.

I've came across a [closed issue](https://github.com/pytorch/pytorch/issues/8653), but I honestly didn't get what was the fix

# Repro

This happens consistently repros on [official detectron2 pipeline](https://app.circleci.com/pipelines/github/facebookresearch/detectron2/2237/workflows/d31b3049-f425-4b31-b103-66c2733930b4/jobs/9902), but I cannot repro this locally.

The original failure happened without using `flags()`, but even after wrapping the unit test within `flags()` ctx mgr as described by the original runtime error message, the failure persisted. It also also attempted `flags(enabled=True)` and `flags(enabled=True, benchmark=True)`, but the same runtime error persisted.

Interesting to note that the possibly offending code is in one unittest file while the triggering code is in another unittest file.
Probably pytest is loading both in memory, creating such broken environment.

Steps to repro

## tests/test_onnxruntime.py (file being added to detectron2 repo gthrough the PR)
```
from torch.testing._internal import common_utils  # Loading module in this file affects other test files
#  (...)
import torch

# PyTorch's `common_utils` is loaded to reference `TestCase`, forcing
#   execution of `torch.backends.disable_global_flags()` in global scope
class TestONNXRuntime(common_utils.TestCase):
    def test_detectron2_mask_rcnnfpn(self, batch_size=2):
        pass
```

## tests/test_engine.py (existing/unchanged test file)
```
import unittest
import torch
from detectron2.engine import DefaultTrainer, SimpleTrainer, default_setup, hooks
#  (...)
# `test_setup_config` fails even with `torch.backends.cudnn.flags()` context manager
#   as suggested by `RuntimeError: not allowed to set torch.backends.cudnn flags after disable_global_flags; please use flags() context manager instead` 
class TestTrainer(unittest.TestCase):
    #  (...)
    def test_setup_config(self):
        with tempfile.TemporaryDirectory(
            prefix="detectron2_test"
        ) as d, torch.backends.cudnn.flags():
            cfg = get_cfg()
            cfg.OUTPUT_DIR = os.path.join(d, "yacs")
            default_setup(cfg, {})  # <-- Failure is triggered here
```

## Running the test

It seems `pytest` loads all test files in memory, sharing the interpreter across all of them. Once the unchanged `torch.testing._internal.common_utils` is loaded by `tests/test_onnxruntime.py` during test collection, the same interpreter is shared with `tests/test_engine.py` (guessing that part).

So at this point, before any test s ran, CUDNN was disabled by PyTorch's `torch.testing._internal.common_utils`'s call to `torch.backends.disable_global_flags()`. When `test_engine.py` runs `default_setup()`, it tries to enable CUDNN and the runtime error is raised, with or without `flags()`

```bash
pytest -sv tests/
```

## Full error message

The full failure is:

```
=================================== FAILURES ===================================
________________________ TestTrainer.test_setup_config _________________________

self = <tests.test_engine.TestTrainer testMethod=test_setup_config>

    def test_setup_config(self):
        with tempfile.TemporaryDirectory(
            prefix="detectron2_test"
        ) as d, torch.backends.cudnn.flags():
            cfg = get_cfg()
            cfg.OUTPUT_DIR = os.path.join(d, "yacs")
>           default_setup(cfg, {})

tests/test_engine.py:193: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
detectron2/engine/defaults.py:226: in default_setup
    cfg, "CUDNN_BENCHMARK", "train.cudnn_benchmark", default=False
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <torch.backends.ContextProp object at 0x7fef8d245e90>
obj = <module 'torch.backends.cudnn' from '/opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages/torch/backends/cudnn/__init__.py'>
val = False

    def __set__(self, obj, val):
        if not flags_frozen():
            self.setter(val)
        else:
            raise RuntimeError("not allowed to set %s flags "
>                              "after disable_global_flags; please use flags() context manager instead" % obj.__name__)
E           RuntimeError: not allowed to set torch.backends.cudnn flags after disable_global_flags; please use flags() context manager instead
```

Looked through [opened issue] https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+torch.backends.cudnn.flags), but couldnt find the exact same thing, only some similar reports from a couple years ago

### Versions

Copied the environemnt from detectron2 the best I could

```
#!/bin/bash -eo pipefail
# disable crash coredump, so unittests fail fast
sudo systemctl stop apport.service
# install from github to get latest; install iopath first since fvcore depends on it
pip install --progress-bar off -U 'git+https://github.com/facebookresearch/iopath'
pip install --progress-bar off -U 'git+https://github.com/facebookresearch/fvcore'
# Don't use pytest-xdist: cuda tests are unstable under multi-process workers.
pip install --progress-bar off ninja opencv-python-headless pytest expecttest tensorboard parameterized pycocotools onnx hypothesis
if [[ "$PYTORCH_VERSION" == "master" ]]; then
  echo "Installing torch/torchvision from $PYTORCH_INDEX"
  # Remove first, in case it's in the CI cache
  pip uninstall -y torch torchvision onnxruntime onnxruntime-training
  pip install -v --no-cache --progress-bar off --pre torch torchvision --extra-index-url $PYTORCH_INDEX
  pip install -v --progress-bar off omegaconf==2.1.2   # needed by FCOSE2ETest::test_empty_data
  if [[ "$ONNXRUNTIME_INDEX" == *"cpu"* || "$PYTORCH_INDEX" == *"cpu"* ]]; then
    echo "CPU-only target detected!"
    export CUDA_HOME="/dev/null"
    export USE_CUDA=0
    export CUDA_VISIBLE_DEVICES=""
  else
    echo "GPU-capable target detected!"
  fi
  pip install --no-cache onnxruntime-training -f $ONNXRUNTIME_INDEX
  python -m onnxruntime.training.ortmodule.torch_cpp_extensions.install
else
  echo "Installing torch==$PYTORCH_VERSION and torchvision==$TORCHVISION_VERSION from $PYTORCH_INDEX"
  pip install -v --progress-bar off torch==$PYTORCH_VERSION torchvision==$TORCHVISION_VERSION -f $PYTORCH_INDEX
fi

python -c 'import torch; print("PyTorch Version:", torch.__version__)'
python -c 'import torchvision; print("TorchVision Version:", torchvision.__version__)'
python -c 'import torch; print("CUDA:", torch.cuda.is_available())'
echo "Python packages"
python -c "import sys; print(sys.executable)"
python --version
pip list
echo "GCC Compiler"
gcc --version
echo "OS Environment Variables"
env

Collecting git+https://github.com/facebookresearch/iopath
  Cloning https://github.com/facebookresearch/iopath to /tmp/pip-req-build-bzm2lbb_
  Running command git clone --filter=blob:none --quiet https://github.com/facebookresearch/iopath /tmp/pip-req-build-bzm2lbb_
  Warning: Permanently added the ECDSA host key for IP address '140.82.113.4' to the list of known hosts.
  Resolved https://github.com/facebookresearch/iopath to commit 4e76b4b597651ef36fd0de98ba104b095d6ad86e
  Preparing metadata (setup.py) ... - done
Requirement already satisfied: tqdm in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from iopath==0.1.10) (4.64.0)
Requirement already satisfied: typing_extensions in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from iopath==0.1.10) (4.3.0)
Requirement already satisfied: portalocker in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from iopath==0.1.10) (2.5.1)
Building wheels for collected packages: iopath
  Building wheel for iopath (setup.py) ... - done
  Created wheel for iopath: filename=iopath-0.1.10-py3-none-any.whl size=31530 sha256=02d48cad272a9bff9fabcf0a9de6fb4a0e40bd3e6c7c6b18a88baaeea1f85c60
  Stored in directory: /tmp/pip-ephem-wheel-cache-zzti8_6w/wheels/0a/c6/dc/e6d2fd4aeccde79a9a9f102519d08ff2a2817672fd7ca92371
Successfully built iopath
Installing collected packages: iopath
  Attempting uninstall: iopath
    Found existing installation: iopath 0.1.9
    Uninstalling iopath-0.1.9:
      Successfully uninstalled iopath-0.1.9
Successfully installed iopath-0.1.10
Collecting git+https://github.com/facebookresearch/fvcore
  Cloning https://github.com/facebookresearch/fvcore to /tmp/pip-req-build-vl7pws80
  Running command git clone --filter=blob:none --quiet https://github.com/facebookresearch/fvcore /tmp/pip-req-build-vl7pws80
  Resolved https://github.com/facebookresearch/fvcore to commit 15171b62a2ee8e84546af8865f32264739234899
  Preparing metadata (setup.py) ... - done
Requirement already satisfied: numpy in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from fvcore==0.1.5) (1.21.6)
Requirement already satisfied: yacs>=0.1.6 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from fvcore==0.1.5) (0.1.8)
Requirement already satisfied: pyyaml>=5.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from fvcore==0.1.5) (6.0)
Requirement already satisfied: tqdm in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from fvcore==0.1.5) (4.64.0)
Requirement already satisfied: termcolor>=1.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from fvcore==0.1.5) (1.1.0)
Requirement already satisfied: Pillow in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from fvcore==0.1.5) (9.2.0)
Requirement already satisfied: tabulate in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from fvcore==0.1.5) (0.8.10)
Requirement already satisfied: iopath>=0.1.7 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from fvcore==0.1.5) (0.1.10)
Requirement already satisfied: typing-extensions in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from iopath>=0.1.7->fvcore==0.1.5) (4.3.0)
Requirement already satisfied: portalocker in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from iopath>=0.1.7->fvcore==0.1.5) (2.5.1)
Requirement already satisfied: ninja in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (1.10.2.3)
Requirement already satisfied: opencv-python-headless in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (4.6.0.66)
Requirement already satisfied: pytest in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (7.1.2)
Collecting expecttest
  Downloading expecttest-0.1.3-py3-none-any.whl (6.5 kB)
Requirement already satisfied: tensorboard in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (2.9.1)
Collecting parameterized
  Downloading parameterized-0.8.1-py2.py3-none-any.whl (26 kB)
Requirement already satisfied: pycocotools in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (2.0.4)
Requirement already satisfied: onnx in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (1.12.0)
Collecting hypothesis
  Downloading hypothesis-6.52.3-py3-none-any.whl (389 kB)
Requirement already satisfied: numpy>=1.14.5 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from opencv-python-headless) (1.21.6)
Requirement already satisfied: iniconfig in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pytest) (1.1.1)
Requirement already satisfied: attrs>=19.2.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pytest) (21.4.0)
Requirement already satisfied: tomli>=1.0.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pytest) (2.0.1)
Requirement already satisfied: py>=1.8.2 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pytest) (1.11.0)
Requirement already satisfied: packaging in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pytest) (21.3)
Requirement already satisfied: importlib-metadata>=0.12 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pytest) (4.12.0)
Requirement already satisfied: pluggy<2.0,>=0.12 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pytest) (1.0.0)
Requirement already satisfied: werkzeug>=1.0.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (2.1.2)
Requirement already satisfied: protobuf<3.20,>=3.9.2 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (3.19.4)
Requirement already satisfied: grpcio>=1.24.3 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (1.47.0)
Requirement already satisfied: setuptools>=41.0.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (47.1.0)
Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (0.6.1)
Requirement already satisfied: absl-py>=0.4 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (1.1.0)
Requirement already satisfied: google-auth<3,>=1.6.3 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (2.9.1)
Requirement already satisfied: requests<3,>=2.21.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (2.28.1)
Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (1.8.1)
Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (0.4.6)
Requirement already satisfied: markdown>=2.6.8 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (3.4.1)
Requirement already satisfied: wheel>=0.26 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from tensorboard) (0.37.1)
Requirement already satisfied: matplotlib>=2.1.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pycocotools) (3.5.2)
Requirement already satisfied: typing-extensions>=3.6.2.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from onnx) (4.3.0)
Collecting sortedcontainers<3.0.0,>=2.1.0
  Downloading sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)
Collecting exceptiongroup>=1.0.0rc8
  Downloading exceptiongroup-1.0.0rc8-py3-none-any.whl (11 kB)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from google-auth<3,>=1.6.3->tensorboard) (5.2.0)
Requirement already satisfied: six>=1.9.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from google-auth<3,>=1.6.3->tensorboard) (1.15.0)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from google-auth<3,>=1.6.3->tensorboard) (0.2.8)
Requirement already satisfied: rsa<5,>=3.1.4 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from google-auth<3,>=1.6.3->tensorboard) (4.8)
Requirement already satisfied: requests-oauthlib>=0.7.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard) (1.3.1)
Requirement already satisfied: zipp>=0.5 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from importlib-metadata>=0.12->pytest) (3.4.0)
Requirement already satisfied: cycler>=0.10 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from matplotlib>=2.1.0->pycocotools) (0.11.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from matplotlib>=2.1.0->pycocotools) (1.4.4)
Requirement already satisfied: fonttools>=4.22.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from matplotlib>=2.1.0->pycocotools) (4.34.4)
Requirement already satisfied: pillow>=6.2.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from matplotlib>=2.1.0->pycocotools) (9.2.0)
Requirement already satisfied: pyparsing>=2.2.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from matplotlib>=2.1.0->pycocotools) (3.0.9)
Requirement already satisfied: python-dateutil>=2.7 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from matplotlib>=2.1.0->pycocotools) (2.8.2)
Requirement already satisfied: idna<4,>=2.5 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from requests<3,>=2.21.0->tensorboard) (3.3)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from requests<3,>=2.21.0->tensorboard) (1.26.10)
Requirement already satisfied: certifi>=2017.4.17 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from requests<3,>=2.21.0->tensorboard) (2022.6.15)
Requirement already satisfied: charset-normalizer<3,>=2 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from requests<3,>=2.21.0->tensorboard) (2.1.0)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard) (0.4.8)
Requirement already satisfied: oauthlib>=3.0.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard) (3.2.0)
Installing collected packages: sortedcontainers, parameterized, expecttest, exceptiongroup, hypothesis
Successfully installed exceptiongroup-1.0.0rc8 expecttest-0.1.3 hypothesis-6.52.3 parameterized-0.8.1 sortedcontainers-2.4.0
Installing torch==1.9+cu111 and torchvision==0.10+cu111 from https://download.pytorch.org/whl/torch_stable.html
Using pip 22.2 from /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages/pip (python 3.7)
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Requirement already satisfied: torch==1.9+cu111 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (1.9.0+cu111)
Requirement already satisfied: torchvision==0.10+cu111 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (0.10.0+cu111)
Requirement already satisfied: typing-extensions in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from torch==1.9+cu111) (4.3.0)
Requirement already satisfied: pillow>=5.3.0 in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from torchvision==0.10+cu111) (9.2.0)
Requirement already satisfied: numpy in /opt/circleci/.pyenv/versions/3.7.9/lib/python3.7/site-packages (from torchvision==0.10+cu111) (1.21.6)
PyTorch Version: 1.9.0+cu111
TorchVision Version: 0.10.0+cu111
CUDA: True
Python packages
/opt/circleci/.pyenv/versions/3.7.9/bin/python
Python 3.7.9
Package                 Version
----------------------- ------------
absl-py                 1.1.0
antlr4-python3-runtime  4.9.3
appdirs                 1.4.4
attrs                   21.4.0
black                   22.3.0
cachetools              5.2.0
certifi                 2022.6.15
charset-normalizer      2.1.0
click                   8.1.3
cloudpickle             2.1.0
cycler                  0.11.0
distlib                 0.3.1
exceptiongroup          1.0.0rc8
expecttest              0.1.3
fairscale               0.4.6
filelock                3.0.12
fonttools               4.34.4
future                  0.18.2
fvcore                  0.1.5
google-auth             2.9.1
google-auth-oauthlib    0.4.6
grpcio                  1.47.0
hydra-core              1.2.0
hypothesis              6.52.3
idna                    3.3
importlib-metadata      4.12.0
importlib-resources     5.8.0
iniconfig               1.1.1
iopath                  0.1.10
kiwisolver              1.4.4
Markdown                3.4.1
matplotlib              3.5.2
mypy-extensions         0.4.3
ninja                   1.10.2.3
nose                    1.3.7
numpy                   1.21.6
oauthlib                3.2.0
omegaconf               2.2.2
onnx                    1.12.0
opencv-python-headless  4.6.0.66
packaging               21.3
panopticapi             0.1
parameterized           0.8.1
pathspec                0.9.0
pep8                    1.7.1
Pillow                  9.2.0
pip                     22.2
platformdirs            2.5.2
pluggy                  1.0.0
portalocker             2.5.1
protobuf                3.19.4
psutil                  5.9.1
py                      1.11.0
pyasn1                  0.4.8
pyasn1-modules          0.2.8
pycocotools             2.0.4
pydot                   1.4.2
Pygments                2.12.0
pyparsing               3.0.9
pytest                  7.1.2
python-dateutil         2.8.2
PyYAML                  6.0
requests                2.28.1
requests-oauthlib       1.3.1
rsa                     4.8
scipy                   1.7.3
setuptools              47.1.0
Shapely                 1.8.2
six                     1.15.0
sortedcontainers        2.4.0
tabulate                0.8.10
tensorboard             2.9.1
tensorboard-data-server 0.6.1
tensorboard-plugin-wit  1.8.1
termcolor               1.1.0
timm                    0.6.5
tomli                   2.0.1
torch                   1.9.0+cu111
torchvision             0.10.0+cu111
tqdm                    4.64.0
typed-ast               1.5.4
typing_extensions       4.3.0
urllib3                 1.26.10
virtualenv              20.2.2
Werkzeug                2.1.2
wheel                   0.37.1
yacs                    0.1.8
zipp                    3.4.0
GCC Compiler
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

OS Environment Variables
CIRCLE_WORKFLOW_JOB_ID=df62addd-e54c-47a0-b03b-3199da96e064
CIRCLE_BUILD_NUM=9902
MANPATH=/opt/circleci/.nvm/versions/node/v12.18.4/share/man:/usr/local/man:/usr/local/share/man:/usr/share/man
CIRCLE_PR_REPONAME=detectron2
ONNXRUNTIME_INDEX=https://download.onnxruntime.ai/onnxruntime_stable_cu111.html
rvm_bin_path=/opt/circleci/.rvm/bin
CIRCLE_NODE_TOTAL=1
PYENV_ROOT=/opt/circleci/.pyenv
NVM_CD_FLAGS=
GEM_HOME=/opt/circleci/.rvm/gems/ruby-2.7.1
TERM=unknown
SHELL=/bin/bash
rvm_gemset_create_on_use_flag=1
SSH_CLIENT=Set_by_build_agent
IRBRC=/opt/circleci/.rvm/rubies/ruby-2.7.1/.irbrc
NVM_PATH=/opt/circleci/.nvm/versions/node/v12.18.4/lib/node
TORCHVISION_VERSION=0.10+cu111
CIRCLE_INTERNAL_SCRATCH=/tmp/circleci-1156517552
MY_RUBY_HOME=/opt/circleci/.rvm/rubies/ruby-2.7.1
CIRCLE_PROJECT_USERNAME=facebookresearch
rvm_trust_rvmrcs_flag=1
CIRCLE_INTERNAL_TASK_DATA=/tmp/.circleci-task-data-62d96a0374224e1c82b39aa9-0-build
rvm_stored_umask=0022
USER=circleci
NVM_DIR=/opt/circleci/.nvm
SSH2_CLIENT=Set_by_build_agent
CIRCLE_NODE_INDEX=0
SUDO_USER=root
CIRCLE_BRANCH=pull/4205
rvm_install_on_use_flag=1
SUDO_UID=0
rvm_path=/opt/circleci/.rvm
rvm_verify_downloads_flag=1
SSH_AUTH_SOCK=/tmp/circleci-1156517552/ssh_auth_sock
USERNAME=circleci
DETECTRON2_DATASETS=~/.torch/datasets
CIRCLE_PULL_REQUEST=https://github.com/facebookresearch/detectron2/pull/4205
CIRCLE_JOB=linux_cuda111_tests_pytorch1.9
MAVEN_OPTS=-Xmx2048m
PYTHON_VERSION=3.7.9
CIRCLE_PULL_REQUESTS=https://github.com/facebookresearch/detectron2/pull/4205
rvm_prefix=/opt/circleci
CI_PULL_REQUEST=https://github.com/facebookresearch/detectron2/pull/4205
CIRCLE_WORKING_DIRECTORY=~/detectron2
CIRCLE_PR_USERNAME=thiagocrepaldi
PATH=/opt/circleci/.nvm/versions/node/v12.18.4/bin:/opt/circleci/.rvm/gems/ruby-2.7.1/bin:/opt/circleci/.rvm/gems/ruby-2.7.1@global/bin:/opt/circleci/.rvm/rubies/ruby-2.7.1/bin:/home/circleci/.go_workspace/bin:/usr/local/go/bin:/opt/circleci/.pyenv/shims:/opt/circleci/.pyenv/bin:/opt/google/google-cloud-sdk/bin:/usr/local/apache-maven/bin:/home/circleci/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin:/usr/local/gradle-6.3/bin:/opt/circleci/.rvm/bin:/opt/circleci/.rvm/bin
MAIL=/var/mail/circleci
CIRCLE_WORKFLOW_WORKSPACE_ID=d31b3049-f425-4b31-b103-66c2733930b4
CIRCLE_USERNAME=
CIRCLECI_PKG_DIR=/opt/circleci
rvm_loaded_flag=1
NVM_NODEJS_ORG_MIRROR=https://nodejs.org/dist
PWD=/home/circleci/detectron2
PYTORCH_VERSION=1.9+cu111
LANG=en_US.UTF-8
CIRCLE_PR_NUMBER=4205
SSH_ASKPASS=false
CIRCLE_BUILD_URL=https://circleci.com/gh/facebookresearch/detectron2/9902
rvm_version=1.29.10-next (master)
CIRCLE_WORKFLOW_ID=d31b3049-f425-4b31-b103-66c2733930b4
CIRCLECI=true
SUDO_COMMAND=/bin/bash -c /tmp/circleci-machine-agent2462712989/circleci-agent _internal agent-runner --runnerAPIBaseURL=https://runner.circleci.com --allocation=308Q30ME --launchInfo=machine-agent\ version\ 1.0.27892-3d3a954\ (image:\ "ubuntu-1604-cuda-11.1:202012-01",\ public\ ip:\ "35.239.72.222",\ provider:\ "GCE") --maxRunTime=5h0m0s --sshAdvertiseAddr=35.239.72.222:54782 --isMachineProvisioner=true
SHLVL=2
PYENV_SHELL=bash
M2_HOME=/usr/local/apache-maven
HOME=/home/circleci
CIRCLE_SHA1=7d28443055a21dab16b2c077e3c4a7da57819654
CI=true
CIRCLE_REPOSITORY_URL=git@github.com:facebookresearch/detectron2.git
BASH_ENV=/tmp/.bash_env-62d96a0374224e1c82b39aa9-0-build
LOGNAME=circleci
CIRCLE_SHELL_ENV=/tmp/.bash_env-62d96a0374224e1c82b39aa9-0-build
PYTORCH_INDEX=https://download.pytorch.org/whl/torch_stable.html
XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop
GEM_PATH=/opt/circleci/.rvm/gems/ruby-2.7.1:/opt/circleci/.rvm/gems/ruby-2.7.1@global
DBUS_SESSION_BUS_ADDRESS=/dev/null
NVM_BIN=/opt/circleci/.nvm/versions/node/v12.18.4/bin
GOPATH=/home/circleci/.go_workspace:/usr/local/go_workspace
NVM_IOJS_ORG_MIRROR=https://iojs.org/dist
GIT_ASKPASS=echo
SUDO_GID=0
DISPLAY=:99
CIRCLE_PROJECT_REPONAME=detectron2
RUBY_VERSION=ruby-2.7.1
rvm_user_install_flag=0
_=/usr/bin/env
```

cc @csarofeen @ptrblck @xwang233 @mruberry

**Comment by thiagocrepaldi** [comment]
incorrect usage
